packages/react/src/helpers/utils.ts

```diff
@@ -1,8 +1,30 @@
+import isEmpty from 'lodash/isEmpty';
+import isArray from 'lodash/isArray';
+import isObject from 'lodash/isObject';
+
 export const isDevelopment = () => process.env.NODE_ENV !== 'production';
 
+const isEmptyObj = (val: any) => isObject(val) && isEmpty(val);
```

Comment:
Can we narrow the argument of `val` here (same with `isEmptyArr`)?

Reply:
I intentially had it as `any` here, because we don't know the type the context `selector` has chosen. But `isObject`, etc provides TypeScript type guards, so this is type-safe.